noxfile.py

```diff
@@ -0,0 +1,57 @@
+import nox
```

Comment:
What was your reasoning for moving to `nox`? I personally prefer `nox` to `tox` as I think it is more useful for actions outside of testing, but wondering why you made the switch as a part of this PR.

Reply:
Similar reasoning. It is more pythony, and we can run python code inside it to aid testing. Since pydata-sphinx-theme was also using it, so jumped at the opportunity 😬 . Making theme package structures more consistent. But happy to continue with tox, if we conclude its better for this project.

Reply:
Well I am personally a fan of `nox` for the following reasons:

- I find `noxfile.py` is easier to quickly read and understand what's going on, since it looks "like a Python script". This also likely is more accessible to more people that aren't familiar with software development workflows.
- I find it easier to re-use commands (like the install command) since you just write them as python functions

We don't have an official org policy on using tox vs. nox (at least I couldn't find anything in [the development conventions](https://executablebooks.org/en/latest/contributing.html)), so I am fine to try it out here and get feedback from the team over time on whether they like it or not. If worst comes to worst we can always switch back to `tox`.

Reply:
It's not the end of the world, but I am against using nox.
tox is by far the more commonly used package, and declarative configuration is much preferred over python files. Plus I use tox-conda which there is no equivalent plugin system for nox.
tox is the org policy and is implemented in nearly every repo in EBP. nox shouldn't only be used if there is a clear requirement for more flexible configuration.

Reply:
I'd also note that this is undoing #211, where we already had this discussion, and I'd moved the package from nox to tox

Reply:
Maybe this is overkill, but we could support both for a time? As long as we document each it shouldn't be a huge deal, and maybe an opportunity to compare their use a bit. Those files don't change much 🤷.

This feels like one of those things that there is no "right" answer for 🤔. I don't have any problems with tox so if push comes to shove I'm fine using that, but for some reason nox just feels more user friendly to me, I can't put my finger on it.

Reply:
So are you advocating for `conf.py` over `_config.yml` also 😝 
Again, I don't want to bike-shed, it should be what is most helpful for people actively doing the development on the package. But, I would most certainly block the change, if what on any of the repos I am actually actively working on 😬

Reply:
@choldgraf @chrisjsewell Since both seem to do our work just fine. I will just keep tox in this PR, and we can try nox in the later PRs if we want to.

Reply:
I'd suggest keeping this change separate. There's a decent amount of baggage in the nox vs tox topic in this project, and it'd be best to keep it separated.

I'm also on board with nox (see pip's development processes) but that's besides the point.

Reply:
I agree with @pradyunsg - no need to conflate two different changes in one PR that is already touching a lot of stuff.